### DISCUSSION TITLE
Dealing with symmetrical body part errors

### DISCUSSION THREAD

**Discussion by antortjim** [talmolab/sleap#1368] [General]
Dear SLEAP devs

I was wondering if there any tips / best practices on how to minimise or workaround situations where SLEAP makes right vs left errors, such as
1) placing left body part A and right body part A in the same place
2) swapping them 
3) only one is visible and the wrong side is assigned i.e. maybe left is visible and left is estimated to be invisible and right is placed where left should be

See example shot of several instances of 1)
![image](https://github.com/talmolab/sleap/assets/13869571/bb87f713-6cce-472b-b7ce-662c9355321b)

This shot contains an example of 1) and of 3) (rearRightLeg is misplaced on the rearLeftLeg)
![image](https://github.com/talmolab/sleap/assets/13869571/a2e1439b-9533-48c0-a891-2de8d0d7664c)

Will adding more and more labeled data eventually solve this problem?

Thank you!
A

**Comment by talmo** [comment]
Hi @antortjim,

Thanks for reaching out! This is indeed an issue we see occasionally as well, especially on fly data where all the legs look so similar. Some tips:

1. You can try increasing the receptive field size of your model by increasing the max stride (try 32) in the centered instance model. This helps the model be able to reason about the geometric relationship between distal body parts to resolve these ambiguities.
2. If you're not already, set the rotation augmentation to +-180 since this will help expose the mode to more images where the animals are in different configurations.
3. Go through your existing labels again and check for this issue. We often see this happen when there's 1-2 frames in which the right/left were labeled incorrectly, usually because the model messed up and the human annotator didn't notice. Even a couple of frames can really throw off the model since it produces such a strong error gradient. To make matters worse, this behavior will be stochastic since we randomly sample frames to use for validation which make exclude these "problem frames" on some training runs but not others, making it harder to diagnose.
4. Add more labels specifically when these issues happen.

Let us know how that goes and if you have any questions!

Cheers,

Talmo
